parcels/interaction/interactionkernelsoa.py

```diff
@@ -0,0 +1,179 @@
+            kernel = InteractionKernelSOA(self.fieldset, self.ptype, pyfunc=kernel)
+        return kernel.merge(self, InteractionKernelSOA)
+
+    def execute_python(self, pset, endtime, dt):
+        """Performs the core update loop via Python"""
+        if self.fieldset is not None:
+            for f in self.fieldset.get_fields():
+                if type(f) in [VectorField, NestedField, SummedField]:
+                    continue
+                f.data = np.array(f.data)
+
+        for pyfunc in self._pyfunc:
+            pset.compute_neighbor_tree(endtime, dt)
+            active_idx = pset._active_particle_idx
+
+            mutator = defaultdict(lambda: [])
+            for particle_idx in active_idx:
+                p = pset[particle_idx]
+                # Don't use particles that are not started.
+                if (endtime-p.time)/dt <= -1e-7:
+                    continue
+
+                neighbors = pset.neighbors_by_index(particle_idx)
+                try:
+                    res = pyfunc(p, pset.fieldset, p.time, neighbors, mutator)
+                except FieldOutOfBoundError as fse_xy:
+                    res = ErrorCode.ErrorOutOfBounds
+                    p.exception = fse_xy
+                except FieldOutOfBoundSurfaceError as fse_z:
+                    res = ErrorCode.ErrorThroughSurface
+                    p.exception = fse_z
+                except TimeExtrapolationError as fse_t:
+                    res = ErrorCode.ErrorTimeExtrapolation
+                    p.exception = fse_t
+
+                except Exception as e:
+                    res = ErrorCode.Error
+                    p.exception = e
+
+                # (!) InteractionKernels do not advance time or change state
+                if res != StateCode.Success:
+                    logger.warning_once("Some InteractionKernel was not completed succesfully.")
+
+            for particle_idx in active_idx:
+                p = pset[particle_idx]
+                try:
+                    for mutator_func, args in mutator[p.id]:
+                        mutator_func(p, *args)
+                except KeyError:
+                    pass
+
+    def execute(self, pset, endtime, dt, recovery=None, output_file=None, execute_once=False):
+        """Execute this Kernel over a ParticleSet for several timesteps"""
+        pset.collection.state[:] = StateCode.Evaluate
+
+        if abs(dt) < 1e-6 and not execute_once:
+            logger.warning_once("'dt' is too small, causing numerical accuracy limit problems. Please chose a higher 'dt' and rather scale the 'time' axis of the field accordingly. (related issue #762)")
+
+        if recovery is None:
+            recovery = {}
+        elif ErrorCode.ErrorOutOfBounds in recovery and ErrorCode.ErrorThroughSurface not in recovery:
+            recovery[ErrorCode.ErrorThroughSurface] = recovery[ErrorCode.ErrorOutOfBounds]
+        recovery_map = recovery_base_map.copy()
+        recovery_map.update(recovery)
```

Comment:
Here, I'm of a split opinion: Leaving it in doesn't harm, and if (in one of the next steps), error recovery is somewhat re-introduced, I guess that these exact lines will be required.

On the other hand: as the kernel currently doesn't do error recovery, this code part is non-functional, I assume ? If so, @erikvansebille may wanna say what the policy is here - leave it in or leave it out ...

Reply:
That is exactly right. If we decide only add error recovery for InteractionKernels in a later extension, and keep the absence of it as a limitation of this first implementation, I would probably prefer to take it out for now.

Reply:
Removed.